Added settable callback lambda for all mocked send methods in the moc…

bonahona opened the pull request:
…ked producer, for testing purposes.

bonahona committed:
Added settable callback lambda for all mocked send methods in the mocked producer, for testing purposes.

williamhogman reviewed (approved):
LGTM

williamhogman commented on src/main/kotlin/producer/mock/MockProducer.kt:
```diff
@@ -3,24 +3,25 @@ package frans.producer.mock
 ): Producer<K, V> {
     private fun doSend(rec: ProducerRecord<K, V>) =
-        CompletableFuture.supplyAsync { doSendResult }
+        CompletableFuture.supplyAsync { doSendResult }.also { onSend.invoke(rec.value()) }
```
Do you really need to write invoke here?

bonahona committed:
Changes to MockProducer and producer factory in general to fix injection on result.

bonahona committed:
Changes to MockConsumerActor to allow for dynamic adding of messages to a mocked queue.

bonahona committed:
Fixes for active mocked worker.

bonahona committed:
Added possibility to stop a worker.
When starting a worker, it returns an object that can be used to call it to stop.

bonahona committed:
Made Worker inherit closable interface.
Removed stop of thread.

bonahona committed:
Added new MockMessage class for use in when mocking messages in mock consumers.

bonahona committed:
Added convenient companion object methods to generate MockedConsumers.

bonahona committed:
Added helper function to create a factory for a mocked consumer actor.

bonahona committed:
Exposing thread from the consumer actor.

bonahona committed:
Change mocked result and how /when it invokes its callbacks.

bonahona committed:
Added some debug println.

bonahona committed:
Removed thread from MockConsumer actor.

bonahona committed:
More debugging.

bonahona committed:
Changed when messages are run.

bonahona committed:
Removed debug logging.

bonahona committed:
Fixed typo in comments.

bonahona committed:
Added default values for message queue, to make writing tests a tiny bit shorter.

bonahona committed:
Removed extra threads from Active mock consumer.
